src/idl_gen_general.cpp

```diff
@@ -1294,8 +1365,26 @@ class GeneralGenerator : public BaseGenerator {
                  size == SizeOf(field.value.type.base_type))) {
               code += "    " + struct_def.name + ".";
               code += FunctionStart('A') + "dd";
-              code += MakeCamel(field.name) + "(builder, " + field.name;
-              if (!IsScalar(field.value.type.base_type)) code += "Offset";
+              code += MakeCamel(field.name) + "(builder, ";
+
+              if (lang_.language == IDLOptions::kCSharp) {
+                if (IsStruct(field.value.type)) {
+                  // for structs the offset must create inside
+                  if (fieldNames.find(field.value.type.struct_def->name) != fieldNames.end()) {
+                    code += WrapInNameSpace(*field.value.type.struct_def, true);
```

Comment:
This is the only place where your new "force" feature is used. Why is it needed? Can it instead be solved locally?

Reply:
The reason was that I would reuse the WrapInNameSpace(...) code.
But I could also do it locally.